internal/db/collection_define.go

```diff
@@ -139,6 +139,12 @@ func (db *DB) patchCollection(
 	existingColsByName := map[string]client.CollectionVersion{}
 	existingColsByID := map[string]client.CollectionVersion{}
 	for _, col := range existingCols {
+		if col.VersionSources == nil {
+			// JSON patch only allows the `-` array-path to be used on non-nil
+			// arrays, so we assign any previously nil arrays to nil here
```

Comment:
question: Can you confirm that `nil arrays to nil` is what you wanted to say. If so, please try to make this clearer because it looks like you're saying to you are assigning nil when in fact you are assiging a new slice.

Reply:
:grin: Nice spot, this should say `nil arrays to empty`

- [ ] Fix nil-empty comment